command/agent_test.go

```diff
@@ -3204,6 +3206,159 @@ auto_auth {
 	require.Truef(t, found, "unable to find consul-template partial message in logs", runnerLogMessage)
 }
 
+// TestAgent_DeleteAfterVersion_Rendering Validates that Vault Agent
+// can correctly render a secret with delete_after_version set.
+func TestAgent_DeleteAfterVersion_Rendering(t *testing.T) {
```

Comment:
Hopefully needless to say, but this test fails with an older version of consul-template (without the fix), and correctly validates the fix works.